hmda-data-publisher/src/main/scala/hmda/publisher/query/component/PublisherComponent2020.scala

```diff
@@ -0,0 +1,535 @@
+       rateSpread,
+       hoepaStatus,
+       lienStatus,
+       creditScoreApplicant,
+       creditScoreCoApplicant,
+       creditScoreTypeApplicant,
+       creditScoreModelApplicant,
+       creditScoreTypeCoApplicant,
+       creditScoreModelCoApplicant,
+       denialReason1,
+       denialReason2,
+       denialReason3,
+       denialReason4,
+       otherDenialReason,
+       totalLoanCosts,
+       totalPoints,
+       originationCharges,
+      ) <> ((LarPartFour2020.apply _).tupled, LarPartFour2020.unapply)
+
+    def larPartFiveProjection =
+      (discountPoints,
+       lenderCredits,
+       interestRate,
+       paymentPenalty,
+       debtToIncome,
+       loanValueRatio,
+       loanTerm,
+       rateSpreadIntro,
+       baloonPayment,
+       insertOnlyPayment,
+       amortization,
+       otherAmortization,
+       propertyValue,
+       homeSecurityPolicy,
+       landPropertyInterest,
+       totalUnits,
+       mfAffordable,
+       applicationSubmission) <> ((LarPartFive2020.apply _).tupled, LarPartFive2020.unapply)
+
+    def larPartSixProjection =
+      (payable,
+       nmls,
+       aus1,
+       aus2,
+       aus3,
+       aus4,
+       aus5,
+       otheraus,
+       aus1Result,
+       aus2Result,
+       aus3Result,
+       aus4Result,
+       aus5Result,
+       otherAusResult,
+       reverseMortgage,
+       lineOfCredits,
+       businessOrCommercial) <> ((LarPartSix2020.apply _).tupled, LarPartSix2020.unapply)
+
+    def larPartSevenProjection =
+      (conformingLoanLimit,
+       ethnicityCategorization,
+       raceCategorization,
+       sexCategorization,
+       dwellingCategorization,
+       loanProductTypeCategorization,
+       tractPopulation,
+       tractMinorityPopulationPercent,
+       tractMedianIncome,
+       tractOccupiedUnits,
+       tractOneToFourFamilyUnits,
+       tractMedianAge,
+       tractToMsaIncomePercent) <> ((LarPartSeven2020.apply _).tupled, LarPartSeven2020.unapply)
+
+  }
+
+  val larTable2020 = TableQuery[LarTable]
+
+  class LarRepository2020(val config: DatabaseConfig[JdbcProfile])
+      extends TableRepository[LarTable, String] {
+
+    override val table: config.profile.api.TableQuery[LarTable] =
+      larTable2020
+
+    override def getId(row: LarTable): config.profile.api.Rep[Id] =
+      row.lei
+
+    def createSchema() = db.run(table.schema.create)
+    def dropSchema() = db.run(table.schema.drop)
+
+    def insert(ts: LarEntityImpl2020): Future[Int] = {
+      db.run(table += ts)
+    }
+
+    def findByLei(lei: String): Future[Seq[LarEntityImpl2020]] = {
+      db.run(table.filter(_.lei === lei).result)
+    }
+
+    def deleteByLei(lei: String): Future[Int] = {
+      db.run(table.filter(_.lei === lei).delete)
+    }
+    def count(): Future[Int] = {
+      db.run(table.size.result)
+    }
+    def getAllLARs(
+        bankIgnoreList: Array[String],includeQuarterly: Boolean): DatabasePublisher[LarEntityImpl2020] = {
+
+      if(includeQuarterly){
```

Comment:
I'm not happy with this implementation however I'm concerned with the performance hit we run into nesting filter options to consolidate this into a single flow. Once we have larger amounts of quarterly data to test with I'd like to test this out more.